### ISSUE TITLE
Create project in new session no longer works

### ISSUE THREAD

**Issue by hadley** [rstudio/rstudio#2620]
1. Click project | new project
2. Choose an existing directory
3. Click open in new session

A new RStudio instance opens with working directory in my home directory, and no `.Rproj` file is created in either location

**Comment by hadley** [comment]
1.2.500

**Comment by jmcphers** [comment]
Finally got it to repro! An interesting issue. In my testing it seems as though there's some chance that it will open a previous project instead of the one you asked for (it doesn't happen 100% of the time for me but it's relatively likely).

**Comment by ronblum** [comment]
@jmcphers The behavior you're seeing sounds similar to #2203?

**Comment by jmcphers** [comment]
#2203 and #2205 still look fixed to me. It looks like the key to reproducing this is in step 2 of @hadley's repo: you have to pick a directory that doesn't already have an `.Rproj` file. If it does, the project opens correctly, but if it doesn't, one doesn't get created, and then RStudio acts just like it does if you start it without specifying a project.

**Comment by jmcphers** [comment]
@gtritchie This was happening because `findProjectFile` returns true in cases where there's no project file; this causes us to not create one because we think one's already there (and with no project file we don't open the project). 

I think the new behavior is correct (and is consistent with the comment `If none are found, returns false.`) but let me know if this looks wrong to you (I didn't totally follow the intent of your last change https://github.com/rstudio/rstudio/commit/bab6a5ac98247331e471898e19856960c5edea08#diff-bf89be110e85566e4448a2646c9c361f).

**Comment by ronblum** [comment]
Confirmed in 1.2.550 on MacOS.

**Comment by gtritchie** [comment]
@jmcphers Short answer, that's the right fix.

Long answer: that commit was part of this PR: https://github.com/rstudio/rstudio/pull/2072, to fix this issue: https://github.com/rstudio/rstudio/issues/1581

Ironically, that specific commit was a tweak intended to make the code easier to understand, but in doing so I introduced this regression.

I went back and re-tested #1581 with your change and confirmed it still works.
